packages/neuron-wallet/src/services/light-runner.ts

```diff
@@ -0,0 +1,148 @@
+import env from 'env'
+import path from 'path'
+import fs from 'fs'
+import { ChildProcess, spawn } from 'child_process'
+import logger from 'utils/logger'
+import { resetSyncTaskQueue } from 'block-sync-renderer'
+import SettingsService from 'services/settings'
+
+const { app } = env
+
+export enum NetworkType {
+  Light,
+  Full
+}
+
+abstract class CKBRunner {
+  protected constructor() {}
+  protected abstract networkType: NetworkType
+  protected runnerProces: ChildProcess | undefined
+  protected static instance: CKBRunner | undefined
+  static getInstance(): CKBRunner {
+    throw new Error('should be called by instance')
+  }
+
+  protected abstract get ckbBinary(): string
+  abstract start(): Promise<void>
+  platform(): string {
+    switch (process.platform) {
+      case 'win32':
+        return 'win'
+      case 'linux':
+        return 'linux'
+      case 'darwin':
+        return 'mac'
+      default:
+        return ''
+    }
+  }
+
+  async stop() {
+    return new Promise<void>(resolve => {
+      resetSyncTaskQueue.push(false)
+      if (this.runnerProces) {
+        logger.info('Runner:\tkilling node')
+        this.runnerProces.once('close', () => resolve())
+        this.runnerProces.kill('SIGKILL')
+        this.runnerProces = undefined
+      } else {
+        resolve()
+      }
+    })
+  }
+}
+
+export class CKBLightRunner extends CKBRunner {
+  protected networkType: NetworkType = NetworkType.Full
+
+  static getInstance(): CKBLightRunner {
+    if (!CKBLightRunner.instance) {
+      CKBLightRunner.instance = new CKBLightRunner()
+    }
+    return CKBLightRunner.instance as CKBLightRunner
+  }
+
+  protected get ckbBinary() {
```

Comment:
I think it's better to rename it to `binary` because I will refactor `packages/neuron-wallet/src/services/ckb-runner.ts` to extend `Runner`. How about `binary`?